crates/validatron/src/lib.rs

```diff
@@ -1,212 +1,165 @@
-#![allow(clippy::type_complexity)]
+//! validatron is a library for check the correctness of rules over types and subsequent compilation into a single function.
+//!
+//! It check if fields specified in a rules are valid for a given type. Example:
+//!
+//! ```
+//! use validatron::validator::get_valid_rule;
+//!
+//! #[derive(Validatron)]
+//! struct MyStruct {
+//!     my_value: i32
+//! }
+//!
+//! let rule = get_valid_rule::<MyStruct>(
+//!     vec![Field::Simple {
+//!         field_name: "my_value".to_string(),
+//!     }],
+//!     Operator::Relational(RelationalOperator::Equals),
+//!     Match::Value("42".to_string()),
+//! )
+//! .unwrap();
```

Comment:
The first thing someone will think is "why the hell would I write something this verbose". Why don't we include the lalrpop language into validatron? (maybe behind a feature flag) I assumed it was because it was not generic and depending on the pulsar event, but it doesn't seem so.

Than we could add a line to this documentation like:
```
// which is equivalent to writing
// let rule = dsl::ConditionParser::new().parse("Exec", r#"my_value == 42"#).unwrap();                          
```

Reply:
yes, because is not generic, you force the user to use an enum as an entrypoint